### PR TITLE
Fixed call encoding in signature digest

### PR DESCRIPTION
**Pull request by svyatonik**
fixes issue from #690 

I've also swapped signature verification with call decoding - this removes additional requirement && seems more intuitive.

### CODE REVIEW

**Review by tomusdrw — APPROVED** [review]
lgtm! Although it looks weird that we don't need to change any tests? Maybe worth adding some runtime-level tests to make sure the entire integration is working correctly?

File: bin/runtime-common/src/messages.rs [hunk 1]
```diff
@@ -352,9 +352,7 @@ pub mod target {
 
 	impl<B: MessageBridge> From<FromBridgedChainEncodedMessageCall<B>> for Result<CallOf<ThisChain<B>>, ()> {
 		fn from(encoded_call: FromBridgedChainEncodedMessageCall<B>) -> Self {
-			let mut input = &encoded_call.encoded_call[..];
-			let _skipped_length = Compact::<u32>::decode(&mut input).map_err(drop)?;
-			CallOf::<ThisChain<B>>::decode(&mut input).map_err(drop)
+			CallOf::<ThisChain<B>>::decode(&mut &encoded_call.encoded_call[..]).map_err(drop)
```
Comment (HCastano) [inline bin/runtime-common/src/messages.rs:355]: Can you explain the fix for me? Why don't we need to decode the payload twice anymore?

**Review by HCastano — APPROVED** [review]

File: bin/runtime-common/src/messages.rs (hunk 1 above)
Comment (svyatonik) [reply bin/runtime-common/src/messages.rs:355]: `Vec` Idecoding now happens when struct itself is decoded.

**Comment by svyatonik** [comment]
> lgtm! Although it looks weird that we don't need to change any tests? Maybe worth adding some runtime-level tests to make sure the entire integration is working correctly?

Yeah - I forgot to add this check to existing test.

**Pull request merged**